File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] output = new int[2]; //Returned array of 2 numbers.
        HashMap<Integer, Integer> foundSet = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++) {
            if(foundSet.containsKey(target - nums[i])) {
                output[0] = i;
                output[1] = foundSet.get(target - nums[i]);
                return output;
            } else if(foundSet.containsKey(nums[i]) == false) { //Add to set if it doesn't exist.
                foundSet.put(nums[i], i);
            }
        }
        
        return output; //Weird thing at the end because Java moment
    }
}

File: 1-two-sum/1-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        //Approach: Create a map of seen numbers.
        val seen = mutableMapOf<Number, Integer>()
        var result = IntArray(2)
        nums.forEachIndexed{ index, num -> 
            if(seen.containsKey(target - num)) {
                result[0] = index
                result[1] = seen.get(target - num) as Int //Kind of ugly. Maybe theres a better way?
                return result
            } else {
                seen.put(num, index as Integer) //Kind of ugly. Maybe theres a better way?
            }
        }
        return result
    }
}